### PR TITLE
Update contributing tutorial

### LINKED ISSUE #1400: Documentation: Contributing tutorial
## Improvements to Contributing tutorial requested 

* Slight update to "what successful Mage install looks like" 
```
Targets:
  configureAgent       sets up an Azure DevOps agent with Mage and ensures that GOPATH/bin is in PATH.
  ensureMage           Ensure Mage is installed and on the PATH.
  getMixins            Install mixins used by tests and example bundles, if not already installed
  setBinExecutable     Run `chmod +x -R bin`.
  startDocker          Ensure the docker daemon is started and ready to accept connections.
  testE2E              Run end-to-end (e2e) tests
  useXBuildBinaries    Copy the cross-compiled binaries from xbuild into bin.
 ```
  
  current output of mage has changed slightly. 
  
 * The tutorial starts out with installing mage but then doesn't appear to use it throughout (appears to be necessary for running make). Helpful to explain that a little bit. 
  
* Missing link for Mage https://magefile.org/ 
  
* Assumes bash shell. Update to be clear about that.  (Default shell on macos is zsh now) 

* In create a new bundle, not specifying the path of porter so if otherwise installed won't be using what was just built. 
* Error received in running install
```➜  hello-world /tmp/porter/bin/porter install
WARNING: the tag field has been deprecated; please replace with a value for the registry field on the Porter manifest instead
===> See https://porter.sh/author-bundles/#bundle-metadata for more details
installing HELLO...
/cnab/app/run: line 2: /cnab/app/porter-runtime: No such file or directory
Error: 1 error occurred:
	* container exit code: 127, message: <nil>

```
* In change porter, be clear the change in creation of file.
* Update "Run ./bin/porter --name YOURNAME to try out your new command." to "./bin/porter hello --name YOURNAME"
* Next steps: link to contributing guidelines https://github.com/getporter/porter/blob/main/CONTRIBUTING.md at the bottom.

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
* Explain the tests and how to use mage to run them.
* Add details for other shells
* Clarified how to use the local porter build
* Link to contributing guide

https://deploy-preview-1423--porter.netlify.app/contribute/tutorial
https://deploy-preview-1423--porter.netlify.app/quickstart/

# What issue does it fix
Fixes #1400

# Notes for the reviewer
None

# Checklist
- [ ] Unit Tests
- [x] Documentation
- [ ] Schema (porter.yaml)

### CODE REVIEW

**Review by vdice — APPROVED** [review]

**Comment by carolynvs** [comment]
@iennae Would you like to review this?

**Comment by jeremyrickard** [comment]
This LGTM to me.

**Review by jeremyrickard — APPROVED** [review]

File: docs/content/contribute/tutorial.md [hunk 1]
```diff
@@ -183,6 +164,48 @@ the original porter repository and `origin` to refer to your fork:
...
+  <List of available targets>
+```
+
+You know that your $GOPATH/bin is configured correctly if you see a list of mage
+targets.
+
+You can enable tab completion for mage as well, so that you can type 
+`mage t[TAB]` and it will complete it with the name of matching targets.
+
+1. Install bash-completion if it isn't already installed with either `brew install
```
Comment (iennae) [inline docs/content/contribute/tutorial.md:193]: Would this be applicable to windows folks? I didn't that far into windows development so I'm not sure.

File: magefile.go [hunk 2]
```diff
@@ -98,8 +98,8 @@ func porter(args ...string) sh.PreparedCommand {
 
 // Run end-to-end (e2e) tests
 func TestE2E() error {
-	mg.Deps(startLocalDockerRegistry)
-	defer stopLocalDockerRegistry()
+	mg.Deps(StarlDockerRegistry)
```
Comment (iennae) [inline magefile.go:101]: curious about the StarlDockerRegistry name from startLocalDockerRegistry .. Should it be StarLDockerRegistry? Just confused here about the naming. I see that the function is defined below and it matches but I wasn't sure if the L stood for local or if this was something else so the camel casing seems weird to me.

Comment (iennae) [reply magefile.go:101]: Maybe it was meant to be StartDockerRegistry to match up with the StopDockerRegistry..

File: docs/content/contribute/tutorial.md (hunk 1 above)
Comment (carolynvs) [reply docs/content/contribute/tutorial.md:193]: At the moment the tutorial says 

> We are improving our support for building Porter on Windows. In a few weeks, we will have it all working on any Windows shell. For now, if you are on Windows, please use Windows Subsystem for Linux (WSL). The rest of this tutorial assumes that you are inside your WSL distribution when installing prerequisites and executing the commands.

So bash completion is something that we can expect from people using WSL.

File: magefile.go (hunk 2 above)
Comment (carolynvs) [reply magefile.go:101]: Yeah it was a typo. I'll submit a PR to fix that! 🤦‍♀️

File: docs/content/contribute/tutorial.md (hunk 1 above)
Comment (carolynvs) [reply docs/content/contribute/tutorial.md:193]: I have been in the process of making _everything_ about our dev process work for Windows (without WSL) but it is taking time. So when I can, I document how to do things in PowerShell, and eventually we will get there! 😀

File: magefile.go (hunk 2 above)
Comment (carolynvs) [reply magefile.go:101]: https://github.com/getporter/porter/pull/1423

**Pull request merged**